app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettings.kt

```diff
@@ -107,6 +143,74 @@ fun WallpaperSettings(
     }
 }
 
+/**
+ * A group of [Wallpaper]s.
+ *
+ * @property wallpapers The list of [Wallpaper]s in this group.
+ * @property title An optional title for this group.
+ * @property limit The maximum number of wallpapers that will be shown in this group.
+ * @property id A unique ID for this group (uses a generated UUID by default)
+ */
+
+data class WallpaperProviderGroup(
+    var wallpapers: List<Wallpaper>,
+    var title: String? = null,
+    val limit: Int = Integer.MAX_VALUE,
+    val id: String = UUID.randomUUID().toString(),
+)
+
+@Composable
+private fun WallpaperGroups(
+    title: String?,
+    onLearnMoreClick: () -> Unit?,
+) {
+    Column(
+        modifier = Modifier.padding(
+            top = 16.dp,
+            start = 18.dp,
+            end = 18.dp,
+            bottom = 12.dp,
+        ),
+    ) {
+        if (title != "default" && title != "classic-firefox") {
+            Text(
+                text = "Limited edition",
+                modifier = Modifier.padding(bottom = 2.dp),
+                color = FirefoxTheme.colors.textSecondary,
+                fontSize = 14.sp,
+            )
+
+            title.let {
+                Row(modifier = Modifier.fillMaxWidth()) {
+                    if (it != null) {
+                        Text(
+                            text = "$it. ",
```

Comment:
I think we should also introduce an extension function to get a more human friendly version of a collection title. Something like 
```
fun Wallpaper.Collection.getDisplayName(): String = this.name
    .split('-')
    .joinToString(" ") { word -> 
        word.replaceFirstChar { 
            if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() 
        } 
    }
```